### PR TITLE
Reducing Tornjak FE image - version 1.0.2

### PR DESCRIPTION
**Pull request by mamy-CS**
- Prune Dev dependencies
- Use a smaller base image
- Filter through npm packages and keep only needed for production 
- Also addresses https://github.com/spiffe/tornjak/issues/149

### CODE REVIEW

File: Dockerfile.frontend-container [hunk 1]
```diff
@@ -1,15 +1,19 @@
-FROM node:18
+
+FROM node:16-alpine
```
Comment (mrsabath) [inline Dockerfile.frontend-container:2]: What is the reason that we are going to an older version of node?

File: Dockerfile.frontend-container [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
+RUN npm prune --production
 
 # set env variables
 ENV REACT_APP_API_SERVER_URI $REACT_APP_API_SERVER_URI
 ENV REACT_APP_AUTH_SERVER_URI $REACT_APP_AUTH_SERVER_URI
-ARG NODE_OPTIONS=--openssl-legacy-provider
 ENV NODE_OPTIONS $NODE_OPTIONS
-
+ARG NODE_OPTIONS=--openssl-legacy-provider
 EXPOSE 3000
```
Comment (mrsabath) [inline Dockerfile.frontend-container:17]: We should parametrize the port as Env variable, so we can easily change it. It's ok for this PR, but we should open an issue for the future.

File: Dockerfile.frontend-container [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
+
 WORKDIR /usr/src/app
-COPY tornjak-frontend ./
-RUN npm install
+
+COPY tornjak-frontend .
+
+RUN npm install --omit=dev
+
+RUN npm prune --production
```
Comment (mrsabath) [inline Dockerfile.frontend-container:10]: oh, so this is the trick to clean it up? :)

File: tornjak-frontend/src/tables/table-body.tsx [hunk 4]
```diff
@@ -57,7 +57,7 @@ class Body extends React.Component<BodyProp, BodyState> {
                                     cell.value)}
                             </TableCell>
                         ))}
-                        {((this.props.entityType === "Agent" && this.TornjakHelper.checkRolesAdminUser(this.props.globalUserRoles)) || (this.props.entityType && !Auth_Server_Uri)) &&
+                        {((this.props.entityType === "Agent" && this.TornjakHelper.checkRolesAdminUser(this.props.globalUserRoles)) || (this.props.entityType === "Agent" && !Auth_Server_Uri)) &&
```
Comment (mrsabath) [inline tornjak-frontend/src/tables/table-body.tsx:60]: This looks like the fix for the issue #149 if so, we should update the PR with "It resolves issue #149" and if testing successful, please close the issue.

**Review by mrsabath — CHANGES_REQUESTED** [review]
If the new image size is reduced it would be a great improvement over the previous image. Just a new minor nits. 
@maia-iyer if your testing of the image is successful, please approve this PR.

File: Dockerfile.frontend-container (hunk 2 above)
Comment (mamy-CS) [reply Dockerfile.frontend-container:17]: Opened issue https://github.com/spiffe/tornjak/issues/153

File: Dockerfile.frontend-container (hunk 1 above)
Comment (mamy-CS) [reply Dockerfile.frontend-container:2]: Image size is a little less and we don't want the updates on the new release but it's a small difference 847MB for the whole image vs 907. So reverting to node 19.

**Review by maia-iyer — APPROVED** [review]
lgtm

**Review by mrsabath — APPROVED** [review]
Nice! lgtm

**Pull request merged**